### PR TITLE
[virtual_nfts] [3] Adds key and value types and serializers needed for storing NFTs in VirtualMerkle.

### PR DESCRIPTION
**Pull request by rjuang**
This PR adds the different key and value types and associated serializers needed for storing NFTs in VirtualMerkle. This PR also adds google's truth testing framework to pom.xml for test scope.

Related issue: #3089

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/state/virtual/UniqueTokenKey.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+    private static int computeNonZeroBytes(long value) {
+        // The value returned from this will range in [0, 8].
+        if (value == 0) {
+            return 0;
+        }
+
+        // Max value here is (64 - 0)/8 = 8
+        // Min value here is ceil((64 - 63)/8) = 1
+        return (int) Math.ceil((Long.SIZE - Long.numberOfLeadingZeros(value)) / 8D);
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/state/virtual/UniqueTokenKey.java:99]: Leemon prefers we avoid floating-point arithmetic everywhere out of maximum paranoia. 😁 

Would an integral-only equivalent be,
```
final var nonZeroBits = Long.SIZE - Long.numberOfLeadingZeros(value);
return (nonZeroBits / 8) + Math.min(1, nonZeroBits % 8);
```

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

File: hedera-node/src/main/java/com/hedera/services/state/virtual/UniqueTokenKey.java [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        }
+        return Long.compare(this.entityNum, other.entityNum);
+    }
+
+    /* package */ interface ByteConsumer {
+        void accept(byte b) throws IOException;
+    }
+
+    private static byte packLengths(final int upper, final int lower) {
+        return (byte) ((upper << 4) | (lower & 0x0F));
```
Comment (JeffreyDallas) [inline hedera-node/src/main/java/com/hedera/services/state/virtual/UniqueTokenKey.java:120]: do we need to worry about big/small endian ?

Comment (rjuang) [reply hedera-node/src/main/java/com/hedera/services/state/virtual/UniqueTokenKey.java:120]: No. Java is always big-endian.

**Review by JeffreyDallas — APPROVED** [review]

**Pull request merged**